src/test/groovy/com/google/protobuf/gradle/AndroidProjectDetectionTest.groovy

```diff
@@ -13,8 +16,8 @@ import spock.lang.Unroll
  */
 @CompileDynamic
 class AndroidProjectDetectionTest extends Specification {
-  private static final List<String> GRADLE_VERSION = ["5.6"]
-  private static final List<String> ANDROID_PLUGIN_VERSION = ["3.5.0"]
+  private static final List<String> GRADLE_VERSION = [GradleVersion.VERSION_5_6, GradleVersion.VERSION_7_4_2]
```

Comment:
Honestly, I find this sort of thing to be negative value (pure cost). I'd rather not have it. It's like having a constant `COMMA = ","` (which I've seen before in a Util class in production code) which is strictly worse than just using the literal. But I'll let @YifeiZhuang weigh in as well.